### PR TITLE
Refactor links

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
## Description

This PR replaces components in `lib/form/connected-links` to use the new `Link` component in `lib/components/util`.

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [x] Are appropriate Typescript types implemented?

### CODE REVIEW

File: lib/util/itinerary.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ interface LegWithOtp1HailedCar extends Leg {
   hailedCar?: boolean
 }
 
-interface ItineraryWithOtp1HailedCar extends Itinerary {
+export interface ItineraryWithOtp1HailedCar extends Itinerary {
```
Comment (amy-corson-ibigroup) [inline lib/util/itinerary.tsx:22]: Should we rename this since we're using it for non OTP1 trips?

**Review by amy-corson-ibigroup — APPROVED** [review]
Looks good! Thank you for the ts changes

File: lib/util/itinerary.tsx (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/util/itinerary.tsx:22]: I am not sure whether hailed cars were ported to OTP2.

File: lib/components/user/nav-login-button.tsx [hunk 2]
```diff
@@ -87,34 +83,27 @@ const NavLoginButton = ({
             links.map((link, i) => {
               if (link.url.startsWith('http')) {
                 return (
-                  <li>
-                    <UnstyledHelpLink href={link.url} target="_blank">
+                  <li key={link.url}>
+                    <NavUnstyledLink as="a" href={link.url} target="_blank">
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply lib/components/user/nav-login-button.tsx]: Good catch! Updated in https://github.com/opentripplanner/otp-react-redux/pull/1199/commits/3720a29fc796b310d7069c8d8ee7a5152506f208.

File: lib/util/itinerary.tsx (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/util/itinerary.tsx:22]: They were! But the structure is slightly different I believe

**Pull request merged**